pkg/apis/monitoring/v1/types.go

```diff
@@ -1637,6 +1637,20 @@ type AlertmanagerSpec struct {
 	// This is an alpha field and requires enabling StatefulSetMinReadySeconds feature gate.
 	// +optional
 	MinReadySeconds *uint32 `json:"minReadySeconds,omitempty"`
+	// EXPERIMENTAL: globalAlertmanagerConfig specifies the global Alertmanager configuration.
+	// If defined, it takes precedence over the `configSecret` field.
+	// This field may change in future releases.
+	// The specified global alertmanager config will not force add a namespace label in routes and inhibitRules.
+	GlobalAlertmanagerConfig *GlobalAlertmanagerConfig `json:"globalAlertmanagerConfig,omitempty"`
```

Comment:
(nit) I don't think we need the `Global` prefix but using `AlertmanagerConfig` could be confusing with the CRD name. `AlertmanagerConfiguration` might be a good middleground.

```suggestion
	alertmanagerConfiguration *AlertmanagerConfiguration `json:"alertmanagerConfiguration,omitempty"`
```

Reply:
fixed at [903c261](https://github.com/prometheus-operator/prometheus-operator/pull/4220/commits/903c2610d710f4d5c5ac574abfb0bae021a8db36)